admin/replication/ProcessReplicationChanges

```diff
@@ -297,6 +299,21 @@ sub dbmirror2_command {
         $newdata = $json->decode($newdata);
     }
 
+    # Some dbmirror2 packets have an incomplete pending_keys file due to a
+    # bug. If we encounter this, fetch primary keys from the database.
```

Comment:
> Will we still be able to spot this bug when this fallback is in place?

Originally I had a warning message here, but I removed it because logging it for packets that are known to be broken didn't seem helpful (i.e. the user may think something is wrong with their setup, and them reporting the issue wouldn't help because we already know about it and fixed it).

However, I'm now thinking I should add the warning back, because 99.99% of people won't be applying any dbmirror2 packets produced before the schema change (and any produced after shouldn't have this bug).

Reply:
Adding the warning back seems sensible enough to me.

Reply:
Added back the warning message.